ngraph/core/reference/include/ngraph/runtime/reference/gather.hpp

```diff
@@ -27,149 +30,386 @@ namespace ngraph
-                for (size_t i = axis; i < params_ndim; i++)
-                {
-                    params_outer_end_corner[i] = 1;
-                }
-                const Strides params_outer_strides(params_ndim, 1);
-                AxisVector params_outer_axis_order(params_ndim);
-                std::iota(params_outer_axis_order.begin(), params_outer_axis_order.end(), 0);
-                const CoordinateTransform params_outer_transform(params_shape,
-                                                                 params_outer_start_corner,
-                                                                 params_outer_end_corner,
-                                                                 params_outer_strides,
-                                                                 params_outer_axis_order);
-
-                // Create a CoordinateTransform for "indices" that visits only the first element
-                // along inner most axis
-                const Coordinate indices_outer_start_corner(indices_ndim, 0);
-                Coordinate indices_outer_end_corner(indices_shape);
-                if (indices_ndim > 0)
-                {
-                    indices_outer_end_corner[indices_ndim - 1] = 1;
-                }
-                const Strides indices_outer_strides(indices_ndim, 1);
-                AxisVector indices_outer_axis_order(indices_ndim);
-                std::iota(indices_outer_axis_order.begin(), indices_outer_axis_order.end(), 0);
-                const CoordinateTransform indices_outer_transform(indices_shape,
-                                                                  indices_outer_start_corner,
-                                                                  indices_outer_end_corner,
-                                                                  indices_outer_strides,
-                                                                  indices_outer_axis_order);
-
-                // Create an inner CoordinateTransfrom for "out"
-                const size_t out_inner_ndim = out_ndim - axis;
-                const Shape out_inner_shape(out_shape.begin() + axis, out_shape.end());
-                const Coordinate out_inner_start_corner(out_inner_ndim, 0);
-                Coordinate out_inner_end_corner(out_inner_shape);
-                if (indices_ndim > 0)
-                {
-                    out_inner_end_corner[indices_ndim - 1] = 1;
+                    // Create a CoordinateTransform for "out" that visits the outer "axis"
+                    // dimensions
+                    size_t out_ndim = static_cast<size_t>(out_shape.size());
+                    Coordinate out_outer_start_corner(out_ndim, 0);
+                    Coordinate out_outer_end_corner(out_shape);
+                    for (size_t i = axis; i < out_ndim; i++)
+                    {
+                        out_outer_end_corner[i] = 1;
+                    }
```

Comment:
std::fill

Reply:
`v0::gather` will be removed when all will be finish.